enterprise_learner_portal/api/v1/views.py

```diff
@@ -151,3 +154,72 @@ def patch(self, request):
         ).data
 
         return Response(data)
+
+
+class EnterpriseAssignedCoursesView(APIView):
+    """
+    View for returning information about given assigned courses.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+
+    def get(self, request):
+        """
+        Returns details about given assigned courses.
+
+        Example response:
+
+        [
+            {
+                "course_run_id": "course-v1:edX+L153+2T2023",
+                "created": "2023-08-28T13:21:55.913099Z",
+                "start_date": "2023-08-30T13:21:55Z",
+                "end_date": "2023-10-17T13:21:55Z",
+                "display_name": "Works of Ivan Turgenev",
+                "course_run_url": "http://localhost:2000/course/course-v1:edX+L153+2T2023/home",
+                "course_run_status": "completed",
+                "pacing": "instructor",
+                "org_name": "edX"
+            },
+            {
+                "course_run_id": "course-v1:edX+P315+2T2023",
+                "created": "2023-08-28T13:21:46.584659Z",
+                "start_date": "2023-08-30T13:21:46Z",
+                "end_date": "2023-10-17T13:21:46Z",
+                "display_name": "Quantum Entanglement",
+                "course_run_url": "http://localhost:2000/course/course-v1:edX+P315+2T2023/home",
+                "course_run_status": "completed",
+                "pacing": "instructor",
+                "org_name": "edX"
+            }
+        ]
+
+        Query parameters:
+        - `course_ids` (List[str], required): A list of course IDs for which details are requested.
+        - The `course_ids` should be URL-encoded, replacing the '+' sign with '%2B'.
+        Example: course-v1:edX%2BL153%2B2T2023
+        """
+
+        if get_course_overviews is None:
+            raise NotConnectedToOpenEdX(
+                _('To use this endpoint, this package must be '
+                  'installed in an Open edX environment.')
+            )
+
+        course_ids = request.GET.getlist('course_ids')
+
+        if not course_ids:
+            return Response(
+                {'error': 'course_ids must be provided as query parameters'},
+                status=HTTP_400_BAD_REQUEST
+            )
+
+        assigned_courses_overviews = get_course_overviews(course_ids)
```

Comment:
@jajjibhai008 Can you please review this PR? Do we need to get data from federated-content-connector?

Reply:
Yes, I believe you are right. Maybe @jajjibhai008 has a better idea on how to resolve this but one way is to use a logic to similar to https://github.com/openedx/frontend-app-learner-portal-enterprise/pull/761/files#diff-c00f27be83b290e262d1740a29f7d730af16e3e84fbc99be26ff09707c8b0f4eR664-R679

Reply:
@mahamakifdar19 for exec-ed courses we need to get dates (start, end) from the [CourseDetail](https://github.com/edx/federated-content-connector/blob/5bcadc41bfdd41a31ce9488f2274ccbca01dd0f3/federated_content_connector/models.py#L9) model that resides in federated-content-connector. 
You can do something similar implementation here https://github.com/openedx/edx-enterprise/blob/8ce4ce35ff430955a8ee8a5d7b34e3005902c879/enterprise_learner_portal/api/v1/serializers.py#L31
 
And also to calculate the course_run_status of exec-ed courses you need to use this https://github.com/openedx/edx-enterprise/blob/3e091b8f2a1475cbed76a96c6088256c92ed9e47/enterprise_learner_portal/utils.py#L52

[Reason] We have stored the wrong dates for exec-ed courses in our courseOverview model. instead of storing dates from course metadata for EE courses, we are storing course_run dates (start, end ). This is a persisting problem that we are facing.